x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryFolder.java

```diff
@@ -53,7 +53,7 @@ protected PhysicalPlan rule(FilterExec plan) {
                 EsQueryExec exec = (EsQueryExec) plan.child();
                 QueryContainer qContainer = exec.queryContainer();
 
-                Query query = ExpressionTranslators.toQuery(plan.condition());
+                Query query = ExpressionTranslators.toQuery(plan.condition(), qContainer.zoneId());
```

Comment:
I think it's too disruptive to pass the `zoneId` to all translations as the vast majority of the nodes will simply ignore it.
It also forces one to think of an expression in terms of a timezone which is incorrect.
Only some expressions have a zoneId so that information should be localized instead of being passed at a higher level.
In this case, the zoneId should be on the comparison operator (either inferred based on the arguments or extracted from the configuration) and the translator should only extract it from there.